CMakeLists.txt

```diff
@@ -0,0 +1,164 @@
+cmake_minimum_required(VERSION 3.16.2)
+
+project(i3-ipc++ VERSION 0.8.0 LANGUAGES CXX)
+
+include(CMakePackageConfigHelpers)
+
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 17)
+    set(CMAKE_CXX_STANDARD_REQUIRED YES)
+endif()
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+#                              rapidjson                               #
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake/modules/")
+find_package(rapidjson REQUIRED)  # TODO
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+#                               i3-ipc++                               #
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+set(headers
+    include/i3_ipc.hpp
+    include/i3_containers.hpp
+    include/i3_ipc_exception.hpp
+    include/i3_ipc_bad_message.hpp
+    include/i3_ipc_unsupported.hpp
+    include/i3_ipc_invalid_argument.hpp
+)
+add_library(i3-ipc++-obj OBJECT)
+target_sources(i3-ipc++-obj
+    PRIVATE
+        src/i3_ipc/i3_ipc-general.cpp
+        src/i3_ipc/i3_ipc-events.cpp
+        src/i3_ipc/i3_ipc-requests.cpp
+        src/exceptions/i3_ipc_invalid_argument.cpp
+        src/exceptions/i3_ipc_exception.cpp
+        src/exceptions/i3_ipc_bad_message.cpp
+        src/exceptions/i3_ipc_unsupported.cpp
+        src/i3_json_parser/i3_json_parser-events.cpp
+        src/i3_json_parser/i3_json_parser-requests.cpp
+        src/i3_json_parser/i3_json_parser-utility.cpp
+        src/i3_message.cpp
+)
+target_compile_features(i3-ipc++-obj PUBLIC cxx_std_17)
+target_compile_options(i3-ipc++-obj
+    PRIVATE
+        -O2 -Wall -Wextra -Wshadow -Wold-style-cast -Wcast-align -Wunused -Wpedantic -Wconversion
+        -Wsign-conversion -Wdouble-promotion -Wformat=2 -Werror -Wfatal-errors
+)
+target_link_libraries(i3-ipc++-obj PRIVATE "${RAPIDJSON_CXX_FLAGS}")
```

Comment:
I think this may be deleted too once you switch to vanilla RapidJSON find_package